bundle/edu.gemini.lchquery.servlet/src/test/scala/edu/gemini/lchquery/servlet/LchQueryRegexSpec.scala

```diff
@@ -58,7 +57,7 @@ class LchQueryRegexSpec extends Specification with ScalaCheck {
   "toRegEx" should {
     "not match something not in an OR list" in {
       forAll (Gen.nonEmptyListOf(queryGen), queryGen) {
-        (lst, other) => !lst.contains(other) ==> (lst.mkString("|").toRegex.matches(other) should beFalse)
+        (lst, other) => !lst.exists(_.matchesLCH(other)) ==> (lst.mkString("|").matchesLCH(other) should beFalse)
```

Comment:
I suppose this is the substantive change.  Maybe a bit of explanation would help me understand what it is.

Reply:
Ah, sure. The problem before was the use of `contains`, which is case-sensitive on strings, so if we had:

`lst = List("N", "M")` and
`other = "n"`,

then `lst.contains(other)` would be `false`, and hence its negation `true`.

As the ODB matching is case insensitive, the regular expression generated by `lst`, namely `N|M`, would match `n`. Thus, the test case would fail.

Thus, we need to switch to use the regular expression of the ODB browser (i.e. `matchesLCH`) when doing the left side of the implication check on `lst`.

Hopefully I explained that in a way that makes sense?

Reply:
Also, this is a test strictly on the ODB browser regex format, so `contains` was naturally the wrong choice here: both sides of the implication should use the ODB browser regex format to even have the condition make sense.

Reply:
Essentially, the idea here is to test that if something does not match anything in a list of regular expressions, then it does not match the OR combination of all of those regular expressions.